guide/content/en/guide/getting-started.md

```diff
@@ -0,0 +1,96 @@
+# Getting Started
+
+Before we begin, make sure you are running Python 3.7 or higher. Currently, Sanic is works with Python versions 3.7 – 3.11.
+
+## Install
+
+```sh
+pip install sanic
+```
+
+## Hello, world application
+
+.. column::
+
+    If you have ever used one of the many decorator based frameworks, this probably looks somewhat familiar to you.
+
+.. column::
+
+    ```python
+    from sanic import Sanic
+    from sanic.response import text
+
+    app = Sanic("MyHelloWorldApp")
+
+    @app.get("/")
+    async def hello_world(request):
+        return text("Hello, world.")
+    ```
+
+
+
+
+.. info:: Tip
+
+    If you are coming from Flask or another framework, there are a few important things to point out. Remember, Sanic aims for **performance**, **flexibility**, and **ease of use**. These guiding principles have tangible impact on the API and how it works.
+    
+    - Sanic does not use a global context. You do not need to import a `request` object from somewhere else. Instead, that object is passed to your handler.
+    - Route handlers must return a response object. This is a very important distinction. Other frameworks allow you to return a string or a dictionary and they will try to figure out what you meant. Sanic requires you to be explicit about what you want to return, therefore saving valuable time in the request/response cycle.
```

Comment:
This is sort of a FAQ and a turndown for some. I don't think the performance argument is relevant because the check is quite fast. Suggested clarified formatting:
```suggestion
    - Handlers must return a response object rather than bare data. This distinguishes our approach from other frameworks that let you return a string or dictionary, which they convert into html or json with a 200 OK status. By insisting on a response object, we avoid assuming just those two formats and allow for custom headers and status codes like 201 Created, and also enable the support of other formats such as msgpack. To make the process more streamlined for your API, a response middleware can be used to convert any data your handlers return into an appropriate response before it reaches Sanic.
```

The middleware approach works but hasn't ever been quite officially supported. I think we should make it so. On a blueprint level this can come in handy.

Reply:
I just reread what you wrote...

We do have an official pattern for this: https://sanic.dev/en/plugins/sanic-ext/convenience.html#custom-serializer

```python
def message(retval, request, action, status):
    return json(
        {
            "request_id": str(request.id),
            "action": action,
            "message": retval,
        },
        status=status,
    )


@app.post("/<action>")
@serializer(message)
async def do_action(request, action: str):
    return "This is a message"
```

However, I suspect this would cause some typing annotation issues that need to be solved.